### PR TITLE
site: Add Contour<>Kubernetes support matrix

### LINKED ISSUE #1906: site: need to document contour <> k8s compatibility
Similar to https://projectcontour.io/resources/envoy/ we should document which versions of k8s are compatible with which versions of contour.

### LINKED ISSUE #2155: Contour should document the minimum version of kubernetes required
At the moment we give users little guidance on the versions of kuberenetes contour is compatible with. 

For the most part compatibility is derived from the version of the kubernetes client (client-go) that Contour is built against. client-go itself publishes guidance on which versions of kubernetes is it compatible with. 

In addition contour may from time to time require features available in a specific kubernetes version. 

I propose that we add a new compatibility page, similar to the one we publish for contour/envoy compatibility. It should list major/minor contour version against major/minor minimum kubernetes version. 

From the version of the client used for contour 1.0 and 1.1, the minimum kubernetes version is 1.14, assuming that the aforementioned contour versions build against version 1.16 of client-go.

### PR DESCRIPTION
**Pull request by stevesloka**
Fixes #2155 
Fixes #1906 

Adds a Kubernetes<>Contour support matrix following client-go standards. 

Signed-off-by: Steve Sloka <slokas@vmware.com>

### CODE REVIEW

File: site/_resources/kubernetes.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+This page describes the compatibility matrix of Contour and Kubernetes versions.
+
+Contour utilizes [client-go][1] to watch for resources in a Kubernetes cluster.
+Since Kubernetes is backwards compatible with clients, older client-go versions will work with many different Kubernetes cluster versions.
+The `client-go` package includes a [compatibility matrix][2] as to what Kubernetes API versions are supported with the version of client-go.  
+
+## Supported Kubernetes versions
+
+| Kubernetes version | Contour v1.0.0 | Contour v1.0.1 | Contour v1.1.0 |
+| ------------ | :-----------: | :-----------: | :----------: |
```
Comment (jpeach) [inline site/_resources/kubernetes.md]: What does the `:---:` do (sorry, being lazy and not checking)? Is the first field different on purpose?

Looks like this comes from the envoy support table, where it doesn't have any effect that I can see 🤷‍♀

File: site/_resources/kubernetes.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+| ------------ | :-----------: | :-----------: | :----------: | 
+| 1.15.x | Supported | Supported | Supported | 
+| 1.16.x | Supported | Supported | Supported |
+| 1.17.x | Supported | Supported | Supported | 
+| 1.18.x | Not Supported<sup>1</sup>  | Not Supported<sup>1</sup> | Not Supported <sup>1</sup> |
+<br>
+
+### Notes
+
+1. It's important to note that since Contour consumes only stable (`v1`) Kubernetes APIs, most Kubernetes versions most likely will not have issues, however, the client-go package does not guarantee compatibility.
```
Comment (jpeach) [inline site/_resources/kubernetes.md]: I see what you are getting at here, but `Ingress` isn't at v1. Maybe we can just say something like:

```
Contour only consumes a small number of quite stable Kubernetes APIs ... 
```

File: site/_resources/kubernetes.md (hunk 1 above)
Comment (stevesloka) [reply site/_resources/kubernetes.md]: It defines the title row

File: site/_resources/kubernetes.md (hunk 2 above)
Comment (stevesloka) [reply site/_resources/kubernetes.md]: Fixed up

**Comment by stevesloka** [comment]
I also added links to this page as well as the Envoy matrix from the support policy page.

File: site/_resources/kubernetes.md (hunk 1 above)
Comment (stevesloka) [reply site/_resources/kubernetes.md]: https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#tables

Comment (jpeach) [reply site/_resources/kubernetes.md]: Oh it centers the text

https://help.github.com/en/github/writing-on-github/organizing-information-with-tables#formatting-content-within-your-table

**Pull request merged**